lib/python/src/bailo/helper/model.py

```diff
@@ -263,3 +282,175 @@ def __unpack_mc(self, res):
+
+        :param tracking_uri: MLFlow Tracking server URI
+        :param experiment_id: MLFlow Tracking experiment ID
+        :raises ImportError: Import error if MLFlow not installed
+        """
+        if ml_flow == 1:
+            client = mlflow.tracking.MlflowClient(tracking_uri=tracking_uri)
+            runs = client.search_runs(experiment_id)
+
+            for run in runs:
+                data = run.data
+                info = run.info
+                inputs = run.inputs
+
+                artifact_uri = info.artifact_uri
+                run_id = info.run_id
+                status = info.status
+                datasets = inputs.dataset_inputs
+                datasets_str = [x.name for x in datasets]
+
+                artifacts = []
+
+                # MLFlow run must be status FINISHED
+                if status != "FINISHED":
+                    continue
+
+                temp_dir = f"bailo_runs/mlflow_{run_id}"
+                if len(mlflow.artifacts.list_artifacts(artifact_uri=artifact_uri)) > 0:
+                    mlflow.artifacts.download_artifacts(artifact_uri=artifact_uri, dst_path=temp_dir)
+                    artifacts.append(temp_dir)
+
+                self.start_run(is_mlflow=True)
+                self.log_params(data.params)
+                self.log_metrics(data.metrics)
+                self.log_artifacts(artifacts)
+                self.log_dataset("".join(datasets_str))
+                self.run_data["run"] = info.run_id
+        else:
+            raise ImportError("Optional MLFlow dependencies (needed for this method) are not installed.")
+
+    def publish(self, mc_loc: str, run_id: str, semver: str | None = "0.1.0", notes: str | None = ""):
+        """Publishes a given experiments results to the model card.
+
+        :param mc_loc: Location of metrics in the model card (e.g. performance.performanceMetrics)
+        :param run_id: Local experiment run ID to be selected
+        :param semver: Semantic version of release to create (if artifacts present)
+        :param notes: Notes for release, defaults to ""
+
+        ..note:: mc_loc is dependent on the model card schema being used
+        """
+        mc = self.model.model_card
+        if mc is None:
+            raise UnboundLocalError("Model card needs to be populated before publishing an experiment.")
+
+        mc = NestedDict(mc)
+
+        for run in self.raw:
+            if run["run"] == run_id:
+                sel_run = run
+                break
+            else:
+                continue
+        else:
+            raise NameError(f"Run {run_id} does not exist.")
+
+        values = []
+
+        for metric in sel_run["metrics"]:
+            for k, v in metric.items():
+                values.append({"name": k, "value": v})
+
+        # Updating the model card
+        parsed_values = [{"dataset": sel_run["dataset"], "datasetMetrics": values}]
+        mc[tuple(mc_loc.split("."))] = parsed_values
```

Comment:
I'd be tempted to add this funtionality to a protected method on the `NestedDict` class. Instead of here, where it's not very readable or reusable.